torch/_dynamo/variables/builtin.py

```diff
@@ -718,6 +718,15 @@ def call_function(
                 ),
             )
 
+        # Handle `for t in iter(user_obj)`, where user_obj defines a __iter__ method.
+        if self.fn is iter and args and isinstance(args[0], UserDefinedObjectVariable):
```

Comment:
Do you actually need to test for UDOV here, it seems to me that the meaning of `iter` is literally "run the `__iter__` impl" (which is either in Python, or tp_iter).